packages/clay-pagination/stories/index.tsx

```diff
@@ -53,4 +56,79 @@ storiesOf('ClayPagination', module)
 				totalPages={totalPages}
 			/>
 		);
+	})
+	.add('w/ bar', () => {
+		const PaginationBar = ({numberOfItems}: any) => {
+			const [active, setActive] = React.useState(false);
+			const [activePage, setActivePage] = React.useState(1);
+			const [perPage, setPerPage] = React.useState(10);
+
+			const options = [
+				{
+					label: '5',
+					value: 5,
+				},
+				{
+					label: '10',
+					value: 10,
+				},
+				{
+					label: '20',
+					value: 20,
+				},
+				{
+					label: '30',
+					value: 30,
+				},
+				{
+					label: '50',
+					value: 50,
+				},
+			];
+
+			return (
+				<div className="pagination-bar">
+					<ClayDropDown
+						active={active}
+						className="pagination-items-per-page"
+						onActiveChange={newVal => setActive(newVal)}
+						trigger={
+							<ClayButton displayType="unstyled">
+								{`${perPage} entries`}
+								<ClayIcon
+									spritemap={spritemap}
+									symbol="caret-double-l"
+								/>
+							</ClayButton>
+						}
+					>
+						<ClayDropDown.ItemList>
+							{options.map((item, i) => (
+								<ClayDropDown.Item
+									key={i}
+									onClick={() => setPerPage(item.value)}
+								>
+									{`${item.label} items`}
+								</ClayDropDown.Item>
+							))}
+						</ClayDropDown.ItemList>
+					</ClayDropDown>
+
+					<div className="pagination-results">
+						{`Showing ${activePage} to ${Math.ceil(
+							numberOfItems / perPage
+						)} of ${numberOfItems}`}
+					</div>
+
+					<ClayPagination
+						activePage={activePage}
+						onPageChange={page => page && setActivePage(page)}
+						spritemap={spritemap}
+						totalPages={Math.ceil(numberOfItems / perPage)}
+					/>
+				</div>
+			);
+		};
+
+		return <PaginationBar numberOfItems={500} />;
```

Comment:
I might expect it would also be helpful to provide an API that looks like...

```
<PaginationBar
	deltas? // array,
	hrefConstructor? // func,
	onDeltaChange? // func,
	onPageChange? // func,
	selectedPage // number,
	selectedDelta // number,
	size? // 'sm' || 'lg',
	totalItems // number
/>
```

`deltas`: This allows for a custom list of numbers. We would have a default as well
`hrefConstructor`: This gives the user the ability to either create URLs for the pagination items
`onDeltaChange` & `selectedDelta`: This helps enforce a controlled mechanism. Also helpful for the parent component to know what the selected delta is.
`onPageChange` & `selectedPage`: This helps enforce a controlled mechanism. Also helpful for the parent component to know what the page is.
`size`: I think there are two sizes right? Or maybe that is just in faro. 
`totalItems`: I prefer 'total' over 'numberOf' since it is a bit more explicit.

Reply:
Yes, We have three different `sizes`. As documented here: https://next.clayui.com/docs/css/components/pagination.html#sizes but I think that is a good idea include `size` property on low-level :)

I agree with `totalItems` too